File: src/domain/repositories/ILessonRepository.ts
import { Lesson } from '../entities/Lesson';
import { LearningPathNode } from '../entities/LearningPathNode';

export interface ILessonRepository {
  getLessonById(lessonId: string): Promise<Lesson | null>;
  getLessonByCourseAndSkill(courseId: string, skillId: string): Promise<Lesson | null>;
  getBaseRoadmapNodes(courseId: string): Promise<LearningPathNode[]>;
}

File: mobile/lib/domain/repositories/i_lesson_repository.dart
import '../entities/lesson.dart';
import '../entities/learning_path_node.dart';

abstract class ILessonRepository {
  Future<Lesson?> getLessonById(String lessonId);
  Future<Lesson?> getLessonByCourseAndSkill(String courseId, String skillId);
  Future<List<LearningPathNode>> getBaseRoadmapNodes(String courseId);
}
